addons/account_update_tax_tags/wizard/account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,232 @@
+
+    # ==== Compute methods ====
+    @api.depends('company_id')
+    def _compute_date_from(self):
+        for wizard in self:
+            tax_lock_date = self.company_id.tax_lock_date
+            wizard.date_from = tax_lock_date + timedelta(days=1) if tax_lock_date else fields.Date.context_today(self)
+
+    @api.depends('date_from')
+    def _compute_display_lock_date_warning(self):
+        for wizard in self:
+            tax_lock_date = self.company_id.tax_lock_date
+            wizard.display_lock_date_warning = tax_lock_date and wizard.date_from < tax_lock_date
+
+    # ==== Business methods ====
+    def _recompute_tax_audit_string(self, aml_ids):
+        """Taken from #odoo/upgrade account/saas~12.3.1.1/end-20-recompute.py """
+        self.flush()
+        pos_order_condition = """
+            (
+                EXISTS(SELECT id FROM pos_order WHERE pos_order.account_move = aml.move_id)
+                AND move.id IS NULL
+                AND debit > 0
+            )
+        """ if column_exists(self.env.cr, "pos_order", "account_move") else "false"
+        query = f"""
+            WITH tags AS (
+                SELECT 
+                    aml.id,
+                    TRIM(LEADING FROM to_char((CASE WHEN t.tax_negate THEN -1 ELSE 1 END)
+                                             *(CASE WHEN move.tax_cash_basis_rec_id IS NULL AND j.type = 'sale' THEN -1 ELSE 1 END)
+                                             *(CASE WHEN move.tax_cash_basis_rec_id IS NULL AND (move.move_type IN ('in_refund', 'out_refund')
+                                                 OR {pos_order_condition})
+                                             THEN -1 ELSE 1 END)
+                                             * aml.balance,
+                                             '999,999,999,999,999,999,990.99')  -- should be enough, even for IRR
+                     ) AS tag_amount,
+                     cur.symbol AS currency,
+                     cur.position AS cur_pos,
+                     COALESCE(trl.tag_name, t.name) AS name
+                FROM account_move_line aml
+                INNER JOIN account_account_tag_account_move_line_rel t_rel ON t_rel.account_move_line_id = aml.id
+                INNER JOIN account_account_tag t ON t.id = t_rel.account_account_tag_id
+                INNER JOIN account_journal j ON j.id = aml.journal_id
+                INNER JOIN account_move move ON move.id = aml.move_id
+                INNER JOIN res_company c ON aml.company_id = c.id
+                INNER JOIN res_currency cur ON c.currency_id = cur.id
+                LEFT JOIN account_tax_report_line_tags_rel tr ON tr.account_account_tag_id = t.id
+                LEFT JOIN account_tax_report_line trl ON tr.account_tax_report_line_id = trl.id
+                WHERE aml.id IN %s -- FIXME @oco is this really ok ? :/ probably we should "explode" it manually ?
```

Comment:
I don't think it is :/ An IN on a big list of ids is usually bad for performance AFAIK.

But maybe we could do this query directly in the main one, with a JOIN, replacing the part doing the SELECT ARRAY_AGG(....) ? This way, we'd let SQL connect everything together, and I think it'd be much faster